### ISSUE TITLE
Extension stopped working in Firefox 129

### ISSUE THREAD

**Issue by HayashiKaori** [tiansh/ass-danmaku#24]
I upgraded Firefox this morning to v129, but now this awesomely useful extension no longer works.
Could this please be fixed? 
This extension is one of my favorite ones, so i really hope it will be fixed or if someone else will fork it and make it work again 😋

**Comment by maristie** [comment]
I suggest you list websites on which the extension fails such that others can reproduce it.

**Comment by UBTL** [comment]
Greetings, the site that stopped working is NicoNico now they are back online and with another different backend apparently

**Comment by HayashiKaori** [comment]
是B站弹幕下载没有显示下载按钮，以前需要手动下拉弹幕列表，才会出现。
运行环境：
win11
Firefox129.0.2 (64 位）

**Comment by maristie** [comment]
> Greetings, the site that stopped working is NicoNico now they are back online and with another different backend apparently

@UBTL Please refer to https://github.com/tiansh/ass-danmaku/pull/22 but update the URL to `'*://public.nvcomment.nicovideo.jp/v1/threads'`

**Comment by maristie** [comment]
> 是B站弹幕下载没有显示下载按钮，以前需要手动下拉弹幕列表，才会出现。 运行环境： win11 Firefox129.0.2 (64 位）

@HayashiKaori Please try this PR https://github.com/tiansh/ass-danmaku/pull/27

**Comment by UBTL** [comment]
> > Greetings, the site that stopped working is NicoNico now they are back online and with another different backend apparently
> 
> @UBTL Please refer to #22 but update the URL to `'*://public.nvcomment.nicovideo.jp/v1/threads'`

alright thank very much for trying to fix it, however try to change the addon or add the addon manually is such an absolute pain so I would rather wait if you could be kind to update the addon on the Mozilla Addon page if possible

**Comment by maristie** [comment]
> > > Greetings, the site that stopped working is NicoNico now they are back online and with another different backend apparently
> > 
> > 
> > @UBTL Please refer to #22 but update the URL to `'*://public.nvcomment.nicovideo.jp/v1/threads'`
> 
> alright thank very much for trying to fix it, however try to change the addon or add the addon manually is such an absolute pain so I would rather wait if you could be kind to update the addon on the Mozilla Addon page if possible

Agree, it's a pain for me too. However, only the repo owner can merge & publish the updated add-on in Mozilla Add-on page. So unfortunately I can't do anything until @tiansh is back.

**Comment by john61020** [comment]
> > Greetings, the site that stopped working is NicoNico now they are back online and with another different backend apparently
> 
> [@UBTL](https://github.com/UBTL) Please refer to [#22](https://github.com/tiansh/ass-danmaku/pull/22) but update the URL to `'*://public.nvcomment.nicovideo.jp/v1/threads'`

Niconico is not working now, hope can fix it.

**Comment by UBTL** [comment]
the "fix" should have been simple, I never was able to get it working because Firefox is an ultra dogshit browser that is exceedingly ultra annoying to get non "official" addons running even if you are trying to get them loaded by using "developer" settings and it doesnt surprises me, I was only using this piece of shit browser for this addon really, any chromium based browser is so damn easy to make and load your own addons in comparison, maybe in fact one of this days I should make a chromium based addon for it, the videos I wanted the danmaku for I used https://github.com/AlexAplin/nndownload which get regularly updated since a lot of the community that still are into danmaku based sites are mainly chinese so their high priority are chinese based sites like acfun/bilibili and so on so Niconico doesnt get any attention anymore and cant blame them when the own company that still owns it is doing a dogshit job at it (cough cough the past hacking that took literal many months to get the site up again cough cough)

**Comment by UBTL** [comment]
TL;DR:
Give up on ever using this addon again, use what I suggested if you really want the danmaku that bad

**Comment by john61020** [comment]
> TL;DR: Give up on ever using this addon again, use what I suggested if you really want the danmaku that bad

Thank you very much for your suggest. Unfortunately, nndownload is too complicated for me. I don't need the video, just the danmaku, so this addon is the most convenient for me. I don't mind reinstalling and changing the file every time I use it. I really hope someone can fix it.

**Comment by john61020** [comment]
> the "fix" should have been simple, I never was able to get it working because Firefox is an ultra dogshit browser that is exceedingly ultra annoying to get non "official" addons running even if you are trying to get them loaded by using "developer" settings and it doesnt surprises me, I was only using this piece of shit browser for this addon really, any chromium based browser is so damn easy to make and load your own addons in comparison, maybe in fact one of this days I should make a chromium based addon for it, the videos I wanted the danmaku for I used https://github.com/AlexAplin/nndownload which get regularly updated since a lot of the community that still are into danmaku based sites are mainly chinese so their high priority are chinese based sites like acfun/bilibili and so on so Niconico doesnt get any attention anymore and cant blame them when the own company that still owns it is doing a dogshit job at it (cough cough the past hacking that took literal many months to get the site up again cough cough)

I tried nndownload, honestly, the experience was terrible. First, the danmaku I got were in a .json file and I had no idea how to convert it or use it. Then, nndownload couldn't download danmaku for paid videos without paying like this addon does. I really need someone to fix this addon, please....
